src/legacy/ui/public/chrome/api/sub_url_hooks.js

```diff
@@ -47,14 +45,14 @@ export function registerSubUrlHooks(angularModule, internals) {
     }
 
     $rootScope.$on('$locationChangeStart', (e, newUrl) => {
+      const getHash = url => url.split('#')[1] || '';
+
       // This handler fixes issue #31238 where browser back navigation
       // fails due to angular 1.6 parsing url encoded params wrong.
-      const parsedAbsUrl = url.parse($location.absUrl());
-      const absUrlHash = parsedAbsUrl.hash ? parsedAbsUrl.hash.slice(1) : '';
+      const absUrlHash = getHash($location.absUrl());
```

Comment:
This change is required to make dashboard drilldown navigation with dashboard and back button work together. 
Just an enhancement  to current encoding differences workaround.
The change simplifies how the hash is extracted from the url, the problem is that url.parse encodes '`' when doing the parsing and for drilldown case this causes 2 browser history records. 

simple back button scenario for dashboard to dashboard drilldown is covered: https://github.com/elastic/kibana/pull/60087/files#diff-401179577e069eaeb5f20de5ad1ba769R109